cmd/root.go

```diff
@@ -51,6 +61,133 @@ func init() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 }
 
+// isArgTuneable returns if the argument provided is allowed to be modified
+func isArgTunable(arg string) bool {
+	return tuneableArgsWhitelist[arg]
+}
+
+// isArgInUse checks to see if the argument is already in use by the system currently
+func isArgInUse(arg, cmdLinePath string) (bool, error) {
+	if cmdLinePath == "" {
+		cmdLinePath = cmdLineFile
+	}
+	content, err := ioutil.ReadFile(cmdLinePath)
+	if err != nil {
+		return false, err
+	}
+
+	checkable := string(content)
+	if strings.Contains(checkable, arg) {
+		return true, nil
+	}
+	return false, nil
+}
+
+// parseTuningFile parses the kernel argument tuning file
+func parseTuningFile(tuningFilePath, cmdLinePath string) ([]types.TuneArgument, []types.TuneArgument, error) {
+	addArguments := []types.TuneArgument{}
+	deleteArguments := []types.TuneArgument{}
+	if tuningFilePath == "" {
+		tuningFilePath = kernelTuningFile
+	}
+	if cmdLinePath == "" {
+		cmdLinePath = cmdLineFile
+	}
+	// Return fast if the file does not exist
+	if _, err := os.Stat(tuningFilePath); os.IsNotExist(err) {
+		glog.V(2).Infof("no kernel tuning needed as %s does not exist", tuningFilePath)
+		// This isn't an error. Return out.
+		return addArguments, deleteArguments, err
+	}
+	// Read and parse the file
+	file, err := os.Open(tuningFilePath)
+	// Clean up
+	defer file.Close()
```

Comment:
if open fails for any reason, this will panic, this statement should go after the error check below

Reply:
cc @ashcrow @jlebon

Reply:
https://github.com/openshift/pivot/pull/52